auth0/resource_auth0_user.go

```diff
@@ -0,0 +1,142 @@
+package auth0
+
+import (
+	"github.com/hashicorp/terraform/helper/schema"
+	auth0 "github.com/yieldr/go-auth0"
+	"github.com/yieldr/go-auth0/management"
+)
+
+func newUser() *schema.Resource {
+	return &schema.Resource{
+		Create: createUser,
+		Read:   readUser,
+		Update: updateUser,
+		Delete: deleteUser,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"user_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
+					return old == "auth0|"+new
+				},
+			},
+			"connection_name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"email": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"username": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"password": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"phone_number": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"user_metadata": {
+				Type:     schema.TypeMap,
+				Optional: true,
+			},
+			"email_verified": {
+				Type:     schema.TypeBool,
+				Optional: true,
+			},
+			"verify_email": {
+				Type:     schema.TypeBool,
+				Optional: true,
+			},
+			"phone_verified": {
+				Type:     schema.TypeBool,
+				Optional: true,
+			},
+			"app_metadata": {
+				Type:     schema.TypeMap,
```

Comment:
There's an issue with terraform's support for structured data in Maps (https://github.com/hashicorp/terraform/issues/6215).  I wonder if this field should be of `schema.TypeString`, and expect a JSON encoded string for now?

Reply:
Indeed perhaps it would be best to treat this as a JSON encoded string as the contents could be entirely arbitrary.